### PR TITLE
Add a way to download raw json data for Power, Energy and Weather, us…

### PR DESCRIPTION
**Pull request by kevmegforest**
…ing a date range.

Also fixed hourly method (removed hardcoded date)

### CODE REVIEW

**Comment by titilambert** [comment]
Thanks for your PR ! I create an other on your own fork. You can also click on `Allow edits from maintainers.` at the bottom right of this screen

**Comment by kevmegforest** [comment]
Thanks, the cleanup was well needed. 1st time doing python.

**Comment by kevmegforest** [comment]
Fix a little syntax error with the for loop, and it's good.

File: pyhydroquebec/client.py [hunk 1]
```diff
@@ -333,15 +333,58 @@ def _get_hourly_data(self, day_date, p_p_id):
...
+            if contract_url:
+                yield from self._load_contract_page(contract_url)
+
+            if(start_date is None):
+                start_date = datetime.date.today() - datetime.timedelta(days=1)
+            start_date = datetime.datetime.strptime(start_date, '%Y-%m-%d')
+            end_date = datetime.datetime.strptime(end_date, '%Y-%m-%d')
+
+            data = {}
+            dates = [ (start_date + datetime.timedelta(n))
```
Comment (titilambert) [inline pyhydroquebec/client.py]: Could you remove the space between `'[ ('`

File: pyhydroquebec/client.py [hunk 2]
```diff
@@ -333,15 +333,58 @@ def _get_hourly_data(self, day_date, p_p_id):
...
+                      for n in range(int((end_date - start_date).days))]
```
Comment (titilambert) [inline pyhydroquebec/client.py]: Coudl you align the `for` with the `(` above.

**Comment by titilambert** [comment]
This is released in version 2.3.0 !
It should be available in few minutes in pypi...

**Comment by titilambert** [comment]
Thanks again !

**Pull request merged**